config/software/datadog-metro.rb

```diff
@@ -0,0 +1,31 @@
+name "datadog-metro"
+default_version "last-stable"
+github_token = ENV['GITHUB_TOKEN']
+
+env = {
+  "GOROOT" => "/usr/local/go",
+  "GOPATH" => "/var/cache/omnibus/src/datadog-metro"
+}
+
+dependency "libpcap"
+
+if ohai['platform_family'] == 'mac_os_x'
+  env.delete "GOROOT"
+  gobin = "/usr/local/bin/go"
+else
+  gobin = "/usr/local/go/bin/go"
+end
+
+build do
+   ship_license "https://raw.githubusercontent.com/DataDog/go-metro/master/LICENSE"
+   command "mkdir -p /var/cache/omnibus/src/datadog-metro/src/github.com/DataDog", :env => env
+   command "git clone https://#{github_token}:x-oauth-basic@github.com/DataDog/go-metro.git", :env => env, :cwd => "/var/cache/omnibus/src/datadog-metro/src/github.com/DataDog"
+   command "git checkout #{default_version} && git pull", :env => env, :cwd => "/var/cache/omnibus/src/datadog-metro/src/github.com/DataDog/go-metro"
+   command "#{gobin} get -v -d github.com/cihub/seelog", :env => env, :cwd => "/var/cache/omnibus/src/datadog-metro"
+   command "#{gobin} get -v -d github.com/google/gopacket", :env => env, :cwd => "/var/cache/omnibus/src/datadog-metro"
+   command "#{gobin} get -v -d github.com/DataDog/datadog-go/statsd", :env => env, :cwd => "/var/cache/omnibus/src/datadog-metro"
+   command "#{gobin} get -v -d gopkg.in/tomb.v2", :env => env, :cwd => "/var/cache/omnibus/src/datadog-metro"
+   command "#{gobin} get -v -d gopkg.in/yaml.v2", :env => env, :cwd => "/var/cache/omnibus/src/datadog-metro"
```

Comment:
It'd be great to have these dependencies pinned to a version, right now the versions we use depend on the state of the cache of omnibus and we have no control on the version we pull.

We could use something like [glock](https://github.com/robfig/glock) (which we use on our `dd-go` repo). Once we've set up a GLOCKFILE in `go-metro`, fetching the dependencies seems to be as simple as calling `glock sync github.com/DataDog/go-metro`.

Don't really know how complicated `glock` would be to install on the build environments though. What do you think?